ApollosProject/apollos-prototype
# Protects fields person resolver from access by non-current person

## DESCRIPTION

### What does this PR do, or why is it needed?

Currently, if you find a way to query for another user (for example, if you discover their Apollos ID) it's possible to get data on them that shouldn't be available (such as gender, email, and campus). 

This PR does two things to make this possible

1. Adds a helper function that can wrap resolver functions and stop them from returning data for users who aren't the current user. 

2. Caches the current user on the context to avoid duplicate requests.

### What design trade-offs/decisions were made?

n/a

### How do I test this PR?

1. Login as yourself in Graphiql, and find your own personalId
2. Run the following query, and ensure that you can see your user's details, but not Burke's details
```
{
  me: node(id:"MY PERSON ID HERE") {
    ... on  Person{
      id
      firstName
      lastName
      campus {
        name
      }
      gender
      email
    }
  }
  burke:node(id:"Person:9e584307fb4f32582830fea02470af67"){
    ... on  Person{
      id
      firstName
      lastName
      campus {
        name
      }
      gender
      email
    }
  }    
}
```

### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
